pkg/kv/kvserver/loqrecovery/record.go

```diff
@@ -164,3 +167,65 @@ func UpdateRangeLogWithRecovery(
 	}
 	return nil
 }
+
+func writeNodeRecoveryResults(
+	ctx context.Context,
+	writer storage.ReadWriter,
+	result loqrecoverypb.PlanApplicationResult,
+	decomNodeIDs []roachpb.NodeID,
+) {
+	err := storage.MVCCPutProto(ctx, writer, nil, keys.StoreLossOfQuorumRecoveryStatusKey(),
+		hlc.Timestamp{}, hlc.ClockTimestamp{}, nil, &result)
+	if err != nil {
+		log.Error(ctx, "failed to write loss of quorum recovery plan application status")
+	}
+	if len(decomNodeIDs) > 0 {
+		actions := loqrecoverypb.DeferredRecoveryActions{DecommissionedNodeIDs: decomNodeIDs}
```

Comment:
nit: may as well pass in `DeferredRecoveryActions` instead of constructing it here, similarly to `PlanApplicationResult`, since we may want to add other info to it as well.